mergify_engine/rules/__init__.py

```diff
@@ -35,33 +37,44 @@
 LOG = daiquiri.getLogger(__name__)
 
 
+# This help mypy for breaking the recurcive definition
+FakeTreeT = typing.Dict[str, typing.Any]
+
+
 @dataclasses.dataclass
 class RuleCondition:
```

Comment:
It'd be great to add some docstring at some point, this is getting hard to follow.